sklearn/utils/_openmp_helpers.pyx

```diff
@@ -56,9 +62,12 @@ cpdef _openmp_effective_n_threads(n_threads=None, only_physical_cores=True):
         # to exceed the number of cpus.
         max_n_threads = omp_get_max_threads()
     else:
+        if _CPU_COUNTS is None:
+          _CPU_COUNTS = {
+            opc: cpu_count(only_physical_cores=opc) for opc in [True, False]
+          }
```

Comment:
and then
```suggestion
        if len(_CPU_COUNTS) == 0:
            for opc in (True, False):
                _CPU_COUNTS[opc] = cpu_count(only_physical_cores=opc)
```

Reply:
Let's go

Reply:
Actually, defaultdict does not accept a key. So doing it the old way instead.